packages/app-page-builder/src/hooks/useVariable.ts

```diff
@@ -1,20 +1,19 @@
 import { useCallback, useEffect, useRef } from "react";
 import { useUpdateElement } from "~/editor/hooks/useUpdateElement";
 import { useActiveElement } from "~/editor/hooks/useActiveElement";
-import { PbBlockVariable } from "~/types";
 
 export function useVariable<TValue = any>(variableId: string) {
     const [element] = useActiveElement();
     const elementRef = useRef(element);
+    const variableIdRef = useRef(variableId);
     const updateElement = useUpdateElement();
-    const variable = element?.data?.variables?.find(
-        (variable: PbBlockVariable) => variable.id === variableId
-    );
+    const variable = element?.data?.variables?.find(variable => variable.id === variableId);
 
     // We're storing a reference to an element, so we don't have to recreate the `onChange` callback on every update.
     useEffect(() => {
         elementRef.current = element;
-    }, [element]);
+        variableIdRef.current = variableId;
```

Comment:
Hm, could we maybe add a comment here explaining why we're storing `variableId` into a ref? 

It has to do something with recreating the `onChange` callback less frequently?